cupy/linalg/einsum.py

```diff
@@ -323,6 +321,9 @@ def reduced_binary_einsum(arr0, sub0, arr1, sub1, sub_others):
     assert len(set0) == len(sub0), 'operand 0 should be reduced: diagonal'
     assert len(set1) == len(sub1), 'operand 1 should be reduced: diagonal'
 
+    if len(sub0) == 0 or len(sub1) == 0:
+        return arr0 * arr1, sub0 + sub1
```

Comment:
> https://github.com/cupy/cupy/pull/2928/files/f2aca6cd0304ac64ceddcabef149c4812f91f573..d64f7b2593c0f3c90cbd1cbd0df2bfcc587f22b9#r364636078

I'd like to use `*` instead of `cupy.matmul` if `len(contract_dims) == 0`.  It does not need to be done in this PR but may close #2904 even if cuTensor is unavailable.